### PR TITLE
feat(connectors-qa): support manifest-only connectors

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

As we're migrating all our connectors to manifest-only format, we must run connector static checks on them. This pull request adds basic support for `manifest-only` connectors in `connectors-qa`: 
- Tests will run applicable checks
- New check to verify correct source-declarative-manifest is used.

## User Impact

None.

### CODE REVIEW

File: airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/packaging.py [hunk 1]
```diff
@@ -63,6 +63,24 @@ def _run(self, connector: Connector) -> CheckResult:
         return self.create_check_result(connector=connector, passed=True, message="PyPi publishing is enabled")
 
 
+class CheckManifestOnlyConnectorBaseImage(PackagingCheck):
```
Comment (ChristoGrab) [inline airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/packaging.py:66]: @natikgadzhi I think we need to add an instance of this class to the `ENABLED_CHECKS` list at the bottom of this file for it to run?

Comment (natikgadzhi) [reply airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/packaging.py:66]: BREH you're right.

File: airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/metadata.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
-
+from __future__ import annotations
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/metadata.py:2]:
```suggestion
```

Well that's not required, let's drop it.

File: airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/packaging.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
-
+from __future__ import annotations
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/packaging.py:2]:
```suggestion
```

**Review by lmossman — APPROVED** [review]
I'm not super familiar with this area of the code, so you may want a review from someone more familiar.

But everything here looks reasonable to me

**Comment by ChristoGrab** [comment]
Looks like we need to run `connectors-qa generate-documentation docs/contributing-to-airbyte/resources/qa-checks.md` to generate the updated docs

**Comment by natikgadzhi** [comment]
Regenerated the docs, this should get it cleared.

**Pull request merged**